region/lowernorfair/west/Screw Attack Room.json

```diff
@@ -727,97 +705,34 @@
-        {"or": [
-          "ScrewAttack",
-          {"obstaclesCleared": ["A"]}
-        ]}
-      ],
-      "clearsObstacles": ["A"]
-    },
-    {
-      "link": [3, 2],
-      "name": "Power Bombs",
-      "requires": [
-        "h_canNavigateHeatRooms",
-        "h_canUsePowerBombs",
-        {"heatFrames": 300}
-      ],
-      "clearsObstacles": ["A"]
-    },
-    {
-      "link": [3, 2],
-      "name": "Bombs",
-      "requires": [
-        "h_canNavigateHeatRooms",
-        "h_canUseMorphBombs",
-        {"heatFrames": 300}
-      ],
-      "clearsObstacles": ["A"]
     },
     {
       "link": [3, 2],
-      "name": "Come in Getting Blue Speed",
-      "entranceCondition": {
-        "comeInGettingBlueSpeed": {
-          "length": 8,
-          "openEnd": 0
-        }
-      },
+      "name": "Reset Fall Speed (Wave)",
       "requires": [
+        {"obstaclesCleared": ["A"]},
+        {"doorUnlockedAtNode": 2},
+        "canResetFallSpeed",
+        "Wave",
         {"heatFrames": 200}
       ],
-      "clearsObstacles": ["A"],
-      "devNote": "FIXME: Running in is not required.  Entering the room with canBlueSpaceJump could work, for example."
-    },
-    {
-      "link": [3, 2],
-      "name": "Grapple Teleport Extension Clip",
-      "entranceCondition": {
-        "comeInWithGrappleTeleport": {
-          "blockPositions": [[5, 3]]
-        }
-      },
-      "requires": [
-        {"heatFrames": 250},
-        {"or": [
-          "canWalljump",
-          "h_canCrouchJumpDownGrab"
-        ]}
-      ],
-      "note": [
-        "After teleporting, as Samus swings to the right, begin holding down to extend Grapple just as Samus is slightly right of vertical.",
-        "Samus should swing back left and get stuck on the floor one tile to the right of the Grapple attachment point.",
-        "Release Grapple and Samus will clip through the floor."
-      ],
-      "devNote": "FIXME: The walljump or crouchjump are to reach the door which may not actually be required."
+      "unlocksDoors": [
+        {"types": ["ammo"], "requires": ["never"]}
+      ]
     },
     {
       "link": [3, 2],
-      "name": "Stored Moonfall Clip",
-      "entranceCondition": {
-        "comeInWithStoredFallSpeed": {
-          "fallSpeedInTiles": 1
-        }
-      },
+      "name": "Reset Fall Speed (Quick Shot)",
       "requires": [
-        "canPrepareForNextRoom",
-        "canMoonfall",
-        {"or": [
-          "canWalljump",
-          "h_canCrouchJumpDownGrab"
-        ]},
-        {"heatFrames": 170}
+        {"obstaclesCleared": ["A"]},
+        {"doorUnlockedAtNode": 2},
+        "canResetFallSpeed",
+        "canTrickyJump",
+        {"heatFrames": 145}
       ],
-      "note": [
-        "Moonfall to clip through the Bomb blocks",
-        "Break spin with Shot or Angle Aim to avoid clipping all the way to the bottom of the room."
+      "unlocksDoors": [
+        {"types": ["super"], "requires": ["canPrepareForNextRoom"]},
```

Comment:
?  Seems unrelated to anything

Reply:
Ah, I put this on as you'd need to know that the door is green beforehand. If you've never been in the room or have the map and it was green, I don't think you could have enough time to react and would instead fall to 5

Reply:
That makes sense 🧠